### DISCUSSION TITLE
jsonable_encoder does not observe property's pydantic Config

### DISCUSSION THREAD

**Discussion by kfreezen** [fastapi/fastapi#8414] [Questions] [answered]
### First Check

- [X] I added a very descriptive title to this issue.
- [X] I used the GitHub search to find a similar issue and didn't find it.
- [X] I searched the FastAPI documentation, with the integrated search.
- [X] I already searched in Google "How to X in FastAPI" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to FastAPI but to [Pydantic](https://github.com/samuelcolvin/pydantic).
- [X] I already checked if it is not related to FastAPI but to [Swagger UI](https://github.com/swagger-api/swagger-ui).
- [X] I already checked if it is not related to FastAPI but to [ReDoc](https://github.com/Redocly/redoc).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
from datetime import datetime

from pydantic import BaseModel, Field
from fastapi.encoders import jsonable_encoder


class A(BaseModel):
    class Config:
        json_encoders = {datetime: lambda t: round(t.timestamp())}

    date: datetime = Field(default_factory=datetime.now)


class B(BaseModel):
    a: A = Field(default_factory=A)


a = A()
b = B()

# a.dict() and b.dict() both serialize A the same way
print(a.dict())
print(b.dict())

# jsonable_encoder does not serialize both the same way.
print(jsonable_encoder(a))
print(jsonable_encoder(b))
```

### Description

I expect `class A` to be serialized the same way by jsonable_encoder regardless of whether it is a nested property of a jsonified class or not.

### Operating System

macOS

### FastAPI Version

0.87.0

### Python Version

3.10.6

**Comment by kfreezen** [comment]
I'm taking a stab at a PR to fix this. Will update if/when I give up.

**Comment by kfreezen** [comment]
Working on fix in #5690

**Comment by azogue** [comment, marked as answer]
Not sure if this is a bug or a feature 😅 (from `pydantic`!)

AFAIK, the `pydantic` `Config` affects to the class object where it's defined, is not related to the **fields**

And to compare the `jsonable_encoder`, the way to do it is with the `.json()` method of pydantic models, as the customisation is defined for it. 

```python
assert json.dumps(jsonable_encoder(a)) == a.json()
assert json.dumps(jsonable_encoder(b)) == b.json()
```
That is correct, so, ¿where is the issue?

To use the export customisation in another object, the `Config` should be defined in the other object too:

```python
class C(BaseModel):
    a: A = Field(default_factory=A)

    class Config:
        json_encoders = {datetime: lambda t: round(t.timestamp())}

c = C()
assert c.json() == json.dumps(jsonable_encoder(c))
assert isinstance(jsonable_encoder(c)["a"]["date"], int)
```

> a.dict() and b.dict() both serialize A the same way

`.dict()` is not affected with the customisation, as it's for `json_encoders`. It always generates a `datetime` value

> jsonable_encoder does not serialize both the same way.

Nor should do. But serialises _like_ `.json()`, so works as expected 👍

**Comment by kfreezen** [comment]
You know what? I didn’t think to check the output from pydantic. If that’s
the way it is there, maybe I should submit my feature change request there
as well.

This did make me rethink the reason I need this in the first place though.
